src/pages/resetPassword.tsx

```diff
@@ -0,0 +1,138 @@
+import { useContext, useState, useEffect } from "react";
+import { NextRouter, useRouter } from "next/router";
+import { Box, Button, Card, FormHelperText, Stack, Typography } from "@mui/material";
+import { grey } from "@mui/material/colors";
+
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+import { userContext } from "supabase/user_context";
+
+import Background from "@/components/public/Background";
+import PasswordTextField from "@/components/resetPassword/PasswordTextField";
+import { validateConfirmNewPassword } from "@/utilities/validation";
+
+import { validation } from "@/types/Validation";
+import { PagePaths } from "enum/pages";
+import Loading from "@/components/public/Loading";
+import { ResetPassword } from "@/services/Password";
+
+type ResetPassword = {
+  password: string;
+  confirmPassword: string;
+};
+
+export default function Home() {
+  const router: NextRouter = useRouter();
+  const supabaseClient = useSupabaseClient<Database>();
+  const userStatus = useContext(userContext);
+  const [canResetPassword, setCanResetPassword] = useState(false);
+
+  const [newPassword, setNewPassword] = useState<ResetPassword>({
+    password: "",
+    confirmPassword: "",
+  });
+  const [isSubmit, setIsSubmit] = useState<boolean>(false);
+  const [isRequesting, setIsRequesting] = useState<boolean>(false);
+
+  const arePasswordsErr: validation = validateConfirmNewPassword(
+    newPassword.password,
+    newPassword.confirmPassword
+  );
+
+  function handlePasswordChange(
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ): void {
+    setIsSubmit(false);
+    setNewPassword({ ...newPassword, [event.target.name]: event.target.value });
+  }
+
+  async function handleSubmit() {
+    setIsSubmit(true);
+
+    if (!arePasswordsErr.err) {
+      // reset password end point goes here
+      setIsRequesting(true);
+      ResetPassword(newPassword.password, supabaseClient)
+        .then(() => {
+          router.push(PagePaths.successResetPassword);
+        })
+        .catch((err) => {
+          setIsRequesting(false);
+          console.log(err);
+          return;
+        });
+    }
+  }
+
+  useEffect(() => {
+    supabaseClient.auth.onAuthStateChange(async (event, session) => {
+      if (event == "PASSWORD_RECOVERY") {
+        setCanResetPassword(true);
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (router.asPath.indexOf("error") != -1) {
```

Comment:
Can you explain me about this interesting code?

Reply:
In case @Jinnapat didn't see this issue, this is your section tho.

Reply:
Supabase's password recovery link that is sent through email use hash which next router could not parse properly, so I could not use router.query. The easiest way to check whether or not the recovery link has expired is to check is there an "error" substring in the link. Conversely, to check whether the link recovery token is valid, I have to check is there a "access_token" in the link.